spec/controllers/new_administrateur/groupe_instructeurs_controller_spec.rb

```diff
@@ -422,6 +422,33 @@ def remove_instructeur(instructeur)
     end
   end
 
+  describe '#export_groupe_instructeurs' do
+    let!(:gi_1_2) { procedure.groupe_instructeurs.create(label: 'groupe instructeur 1 2') }
+    let!(:instructeur_assigned_1) { create :instructeur, email: 'instructeur_1@ministere_a.gouv.fr', administrateurs: [admin] }
+    let!(:instructeur_assigned_2) { create :instructeur, email: 'instructeur_2@ministere_b.gouv.fr', administrateurs: [admin] }
+
+    subject do
+      get :export_groupe_instructeurs, params: { procedure_id: procedure.id, format: :csv }
+    end
+
+    before do
+      procedure.administrateurs << admin
+      gi_1_2.instructeurs << [instructeur_assigned_1, instructeur_assigned_2]
+    end
+
+    before { Timecop.freeze(Time.zone.local(2021, 1, 2, 23, 11, 14)) }
+    after { Timecop.return }
+
+    let(:procedure) { create(:procedure, :published) }
+
+    it { expect(subject.header["Content-Disposition"]).to include("#{procedure.id}-groupe-instructeurs-#{Date.today}.csv") }
+    it { expect(subject.status).to eq(200) }
+    it { expect(subject.stream.body.split("\n").size).to eq(3) }
+    it { expect(subject.stream.body).to include("groupe instructeur 1 2") }
+    it { expect(subject.stream.body).to include(instructeur_assigned_1.email) }
+    it { expect(subject.stream.body).to include(instructeur_assigned_2.email) }
```

Comment:
Chaque `it` va exécuter le test en entier – y compris la création de tous les objets de mock en base, ce qui prend pas mal de temps.

Pour éviter que les tests prennent trop de temps, j'ai l'impression que c'est bien qu'on regroupe un peu les its qui sont logiquement liés. Ici je dirais qu'on peut tout regrouper :

```ruby
it 'generates a CSV file containing the instructeurs and groups' do
  # …
end
```